native/src/fairseq2n/data/dynamic_bucket_data_source.cc

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+// All rights reserved.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include "fairseq2n/data/dynamic_bucket_data_source.h"
+
+#include <vector>
+
+namespace fairseq2n::detail {
+
+dynamic_bucket_data_source::dynamic_bucket_data_source(
+    std::unique_ptr<data_source> &&inner, float64 threshold, cost_fn &&fn, std::optional<std::size_t> maybe_nb_min, std::optional<std::size_t> maybe_nb_max, bool drop_remainder) noexcept
```

Comment:
Having checked the char length, but if this line (or any other line in your PR) is longer than 100 characters, please do a line break. You can check out other files on how we do it.